waku/waku.go

```diff
@@ -1030,246 +1019,84 @@ func (w *Waku) Stop() error {
 
 // HandlePeer is called by the underlying P2P layer when the waku sub-protocol
 // connection is negotiated.
-func (w *Waku) HandlePeer(peer *p2p.Peer, rw p2p.MsgReadWriter) error {
+func (w *Waku) HandlePeer(p2pPeer *p2p.Peer, rw p2p.MsgReadWriter) error {
 	// Create the new peer and start tracking it
-	wakuPeer := newPeer(w, peer, rw, w.logger.Named("waku/peer"))
+	var peer common.Peer = v0.NewPeer(w, p2pPeer, rw, w.logger.Named("waku/peer"))
```

Comment:
Should we be calling a concrete implementation of the `NewPeer()` function? This could be a good place to implement factory pattern.

Something like:

`var peer common.Peer = PeerFactory(w, p2pPeer, rw, w.logger.Named("waku/peer"))`

Then `PeerFactory()` could also be responsible for finding the highest version of compatibility and passing out the concrete `common.Peer` implementation.

Reply:
sure, makes sense, I will implement that in the same PR as we introduce `v1` code if that's ok